### DISCUSSION TITLE
Some monsters are not mapped in GOAL-API (getAllGoals)

### DISCUSSION THREAD

**Discussion by eerovil** [PaperMC/Paper#10838] [Ideas]
### Is your feature request related to a problem?

I'd like to use the goals api to change mob behaviour. It's working very nicely for the mobs that are mapped. But, for some mobs, such as Piglins, the `Bukkit.getAllGoals()` doesn't return any goals (Not even inherited from Creaturs/Monster). So I can't remove or replace goals.

Looking at this file `VanillaGoal.java`, there seems to be a mapping for crossbow attack, but it's not mapped to any creature.

https://github.com/PaperMC/Paper/blob/master/paper-api-generator/generated/com/destroystokyo/paper/entity/ai/VanillaGoal.java

### Describe the solution you'd like.

Add Default mapping of monster and creture inheritance of all mobs respectively for future-proofing

### Describe alternatives you've considered.

I've looked into workarounds but I'm not savvy enough to get anywhere.

### Other

This is my function to make monsters passive. 

For zombies, skeletons, etc. it works. But here are some mobs that don't:
No goals found for Zoglin
No goals found for Piglin Brute
No goals found for Axolotl
No goals found for Goat
No goals found for Allay

```

    public void makePassive(Monster monster) {
        getLogger().info("Making " + monster.getName() + " id " + monster.getUniqueId() + " passive");
        MobGoals goals = Bukkit.getMobGoals();
        Mob mob = (Mob) monster;
        Collection<Goal<Mob>> monsterGoals = goals.getAllGoals(mob);

        // If no goals found
        if (monsterGoals.isEmpty()) {
            getLogger().info("No goals found for " + monster.getName());
            return;
        }

        monsterGoals.forEach(wrappedGoal -> {
            GoalKey<Mob> key = wrappedGoal.getKey();
            // Only remove keys that have the string attack in key
            List<String> badKeys = List.of("attack", "leap_at");
            // Remove all creeper-specific goals
            List<String> badClasses = List.of("creeper");

            Boolean isBad = badClasses.stream().anyMatch(key.getNamespacedKey().getClass().toString()::contains);
            isBad = isBad || badKeys.stream().anyMatch(key.getNamespacedKey().getKey()::contains);

            if (isBad) {
                goals.removeGoal(monster, key);
            } else {
                // getLogger().info("Keeping goal: " + key.getNamespacedKey().getKey());
            }
        });
    }
    ```

**Comment by lynxplay** [comment]
Duplicate of #10487

**Comment by eerovil** [comment]
Thanks, I did not realize. I also found a workaround for my problem of making these mobs passive: Just cancel the targeting event if the target is a player.
